src/test/scala/org/kframework/minikore/TextToMiniTest.scala

```diff
@@ -61,347 +61,443 @@ class TextToMiniTest {
+            |"""),
           msg)
     }
   }
 
   @Test def parseTestFail14(): Unit = {
     val s =
-      """
+      strip("""
         |[]
         |module _A
         |endmodule
-        |""".stripMargin
+        |""")
     try {
       parseFromString(s)
       assert(false)
     } catch {
       case ParseError(msg) =>
         assertEquals(
-          """
-            |ERROR: Line 3: Column 8: Expected <ModuleName>, but '_'
+          strip("""
+            |ERROR: Line 2: Column 8: Expected <ModuleName>, but '_'
             |module _A
             |       ^
-            |""".stripMargin.stripPrefix(System.lineSeparator()).stripSuffix(System.lineSeparator()),
+            |"""),
           msg)
     }
   }
 
   @Test def parseTestFail15(): Unit = {
     val s =
-      """
+      strip("""
         |[]
         |module A
         |  syntax X ::= x(Y Z) []
         |endmodule
-        |""".stripMargin
+        |""")
     try {
       parseFromString(s)
       assert(false)
     } catch {
       case ParseError(msg) =>
         assertEquals(
-          """
-            |ERROR: Line 4: Column 20: Expected ')' or ',', but 'Z'
+          strip("""
+            |ERROR: Line 3: Column 20: Expected ')' or ',', but 'Z'
             |  syntax X ::= x(Y Z) []
             |                   ^
-            |""".stripMargin.stripPrefix(System.lineSeparator()).stripSuffix(System.lineSeparator()),
+            |"""),
+          msg)
+    }
+  }
+
+  @Test def parseTestFailWhiteSpacesPrefix(): Unit = {
+    val s =
+      """
+
+
+
+        |[]
+        |module A
+        |endmodule [}
+        |""".stripMargin
+    try {
+      parseFromString(s)
+      assert(false)
+    } catch {
+      case ParseError(msg) =>
+        assertEquals(
+          strip("""
+                  |ERROR: Line 7: Column 12: Expected <Symbol> or ']', but '}'
+                  |endmodule [}
+                  |           ^
+                  |"""),
+          msg)
+    }
+  }
+
+  @Test def parseTestFailWhiteSpacesSuffix(): Unit = {
+    val s =
+      """
+        |[]
+        |module A
+        |endmodule [}
+        |
+        |
+        |
+        |""".stripMargin
+    try {
+      parseFromString(s)
+      assert(false)
+    } catch {
+      case ParseError(msg) =>
+        assertEquals(
+          strip("""
+                  |ERROR: Line 4: Column 12: Expected <Symbol> or ']', but '}'
+                  |endmodule [}
+                  |           ^
+                  |"""),
           msg)
     }
   }
 
+  @Test def parseTestFailUnfinishedModule(): Unit = {
+    val s =
+      """
+        |[]
+        |module A
+        |endmodule
+        |""".stripMargin
+    try {
+      parseFromString(s)
+      assert(false)
+    } catch {
+      case ParseError(msg) =>
+        assertEquals(
+          strip("""
+                  | add corresponding error message
+                  |"""),
```

Comment:
Yes, it makes sense to throw an error if the EOF is reached before a complete definition is parsed.  It would also be nice to give precise error messages, as I suggested in the tests, so there is no doubt about what the problem was.